packages/editor/src/components/browser/tableCol/TableColBrowser.tsx

```diff
@@ -1,21 +1,129 @@
-import { useState } from 'react';
-import { Input } from '../../widgets';
+import { useEffect, useMemo, useState } from 'react';
+import { SelectRow, Table, TableCell } from '../../widgets';
 import type { UseBrowserImplReturnValue } from '../useBrowser';
 import { IvyIcons } from '@axonivy/editor-icons';
 import type { BrowserValue } from '../Browser';
+import { useEditorContext, useMeta } from '../../../context';
+import {
+  useReactTable,
+  type ColumnDef,
+  type RowSelectionState,
+  getCoreRowModel,
+  getFilteredRowModel,
+  flexRender
+} from '@tanstack/react-table';
+import { useQueryData } from '../../parts/query/useQueryData';
+import type { DatabaseColumn } from '@axonivy/inscription-protocol';
 export const TABLE_COL_BROWSER_ID = 'tablecol' as const;
 
-export const useTableColBrowser = (): UseBrowserImplReturnValue => {
-  const [value, setValue] = useState<BrowserValue>({ cursorValue: 'table column' });
+export const useTableColBrowser = (onDoubleClick: () => void): UseBrowserImplReturnValue => {
+  const [value, setValue] = useState<BrowserValue>({ cursorValue: '' });
   return {
     id: TABLE_COL_BROWSER_ID,
     name: 'Table Column',
-    content: <TableColumnBrowser value={value.cursorValue} onChange={(change: string) => setValue({ cursorValue: change })} />,
+    content: <TableColumnBrowser value={value.cursorValue} onChange={setValue} onDoubleClick={onDoubleClick} />,
     accept: () => value,
     icon: IvyIcons.Rule
   };
 };
 
-const TableColumnBrowser = (props: { value: string; onChange: (value: string) => void }) => {
-  return <Input {...props} />;
+const TableColumnBrowser = (props: { value: string; onChange: (value: BrowserValue) => void; onDoubleClick: () => void }) => {
+  const { elementContext: context } = useEditorContext();
+  const { config } = useQueryData();
+
+  const columnMetas = useMeta('meta/database/columns', { context, database: config.query.dbName, table: config.query.sql.table }, []).data;
+
+  const [data, setData] = useState<DatabaseColumn[]>([]);
+
+  const [showHelper, setShowHelper] = useState(false);
+
+  useEffect(() => {
+    const select = config.query.sql.select;
+    setData(select.length === 1 && select.indexOf('*') === 0 ? columnMetas : columnMetas.filter(c => select.includes(c.name)));
```

Comment:
instead of `? :` I would write this with if else, IMO much easier to read, and instead of index just compere the first element:

```ts
let columns = columnMetas;
if (select.length > 1 || select[0] !== '*') {
  columns = columnMetas.filter(c => select.includes(c.name)));
}
setData(columns);
```